cluster-autoscaler/cluster_autoscaler.go

```diff
@@ -273,6 +275,24 @@ func run(_ <-chan struct{}) {
 					continue
 				}
 
+				// Check if there are any nodes that failed to register in kuberentes
+				// master.
+				unregisteredNodes := autoscalingContext.ClusterStateRegistry.GetUnregisteredNodes()
+				if len(unregisteredNodes) > 0 {
+					glog.V(1).Infof("%d unregistered nodes present", len(unregisteredNodes))
+					removedAny, err := removeOldUnregisteredNodes(unregisteredNodes, &autoscalingContext, time.Now())
+					// There was a problem with removing unregistered nodes. Retry in the next loop.
+					if err != nil {
```

Comment:
You are not handling correctly the case when some nodes were removed, but error was observed afterwards:  "Some unregistered nodes were removed" should be printed before printing error.